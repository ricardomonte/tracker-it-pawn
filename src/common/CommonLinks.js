import React from 'react';
import { Link, useHistory } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faDog, faWallet, faPlus, faClock, faSignOutAlt,
} from '@fortawesome/free-solid-svg-icons';
import TrackerStyle from '../styles/CommonLinks.module.css';

const CommonLinks = () => {
  const history = useHistory();
  const handleLogOut = () => {
    localStorage.removeItem('token');
    history.push('/');
  };

  return (
    <div className={TrackerStyle.containerLinks}>
      <div>
        <Link className={TrackerStyle.linkDog} to="/add_dog">
          <FontAwesomeIcon icon={faDog} size="3x" />
          <FontAwesomeIcon icon={faPlus} size="2x" className={TrackerStyle.plus} />
        </Link>
      </div>
      <div>
        <Link className={TrackerStyle.linkExp} to="/create_expense">
          <FontAwesomeIcon icon={faWallet} size="3x" />
          <FontAwesomeIcon icon={faPlus} size="2x" className={TrackerStyle.plus} />
        </Link>
      </div>
      <div>
        <Link className={TrackerStyle.linkExp} to="/create_hour">
          <FontAwesomeIcon icon={faClock} size="3x" />
          <FontAwesomeIcon icon={faPlus} size="2x" className={TrackerStyle.plus} />
        </Link>
      </div>
      <div className={TrackerStyle.logOut}>
        <FontAwesomeIcon icon={faSignOutAlt} onClick={handleLogOut} role="presentation" size="3x" />
      </div>
    </div>
  );
};

export default CommonLinks;
