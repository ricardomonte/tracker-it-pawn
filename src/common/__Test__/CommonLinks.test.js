import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import CommonLinks from '../CommonLinks';

test('Content from CommonLinks matches snapshot', () => {
  const content = render(
    <BrowserRouter>
      <CommonLinks />
    </BrowserRouter>,
  );
  expect(content.container).toMatchSnapshot();
});
