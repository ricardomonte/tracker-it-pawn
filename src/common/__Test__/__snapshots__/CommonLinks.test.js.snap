// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content from About matches snapshot 1`] = `
<div>
  <div
    class="containerLinks"
  >
    <div>
      <a
        class="linkDog"
        href="/add_dog"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-dog fa-w-18 fa-3x "
          data-icon="dog"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M298.06,224,448,277.55V496a16,16,0,0,1-16,16H368a16,16,0,0,1-16-16V384H192V496a16,16,0,0,1-16,16H112a16,16,0,0,1-16-16V282.09C58.84,268.84,32,233.66,32,192a32,32,0,0,1,64,0,32.06,32.06,0,0,0,32,32ZM544,112v32a64,64,0,0,1-64,64H448v35.58L320,197.87V48c0-14.25,17.22-21.39,27.31-11.31L374.59,64h53.63c10.91,0,23.75,7.92,28.62,17.69L464,96h64A16,16,0,0,1,544,112Zm-112,0a16,16,0,1,0-16,16A16,16,0,0,0,432,112Z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-plus fa-w-14 fa-2x plus"
          data-icon="plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            fill="currentColor"
          />
        </svg>
      </a>
    </div>
    <div>
      <a
        class="linkExp"
        href="/create_expense"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-wallet fa-w-16 fa-3x "
          data-icon="wallet"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M461.2 128H80c-8.84 0-16-7.16-16-16s7.16-16 16-16h384c8.84 0 16-7.16 16-16 0-26.51-21.49-48-48-48H64C28.65 32 0 60.65 0 96v320c0 35.35 28.65 64 64 64h397.2c28.02 0 50.8-21.53 50.8-48V176c0-26.47-22.78-48-50.8-48zM416 336c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-plus fa-w-14 fa-2x plus"
          data-icon="plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            fill="currentColor"
          />
        </svg>
      </a>
    </div>
    <div>
      <a
        class="linkExp"
        href="/create_hour"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-clock fa-w-16 fa-3x "
          data-icon="clock"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-plus fa-w-14 fa-2x plus"
          data-icon="plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            fill="currentColor"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
